server/channels/app/draft_test.go

```diff
@@ -303,11 +303,47 @@ func TestGetDraftsForUser(t *testing.T) {
 		assert.Equal(t, draftWithFiles.ChannelId, draftsWithFilesResp[0].ChannelId)
 		assert.ElementsMatch(t, draftWithFiles.FileIds, draftsWithFilesResp[0].FileIds)
 
+		assert.Len(t, draftsWithFilesResp[0].Metadata.Files, 1)
 		assert.Equal(t, fileResp.Name, draftsWithFilesResp[0].Metadata.Files[0].Name)
 
 		assert.Len(t, draftsWithFilesResp, 2)
 	})
 
+	t.Run("get draft with invalid files", func(t *testing.T) {
+		// upload file
+		sent, readFileErr := testutils.ReadTestFile("test.png")
+		require.NoError(t, readFileErr)
+
+		fileResp1, updateDraftErr := th.App.UploadFileForUserAndTeam(th.Context, sent, channel.Id, "test1.png", user.Id, "")
+		assert.Nil(t, updateDraftErr)
+
+		fileResp2, updateDraftErr := th.App.UploadFileForUserAndTeam(th.Context, sent, channel.Id, "test2.png", th.BasicUser2.Id, "")
+		assert.Nil(t, updateDraftErr)
+
+		draftWithFiles := draft1
+		draftWithFiles.FileIds = []string{fileResp1.Id, fileResp2.Id}
+
+		draftResp, updateDraftErr := th.App.UpsertDraft(th.Context, draft1, "")
+		assert.Nil(t, updateDraftErr)
+
+		assert.Equal(t, draftWithFiles.Message, draftResp.Message)
+		assert.Equal(t, draftWithFiles.ChannelId, draftResp.ChannelId)
+		assert.ElementsMatch(t, draftWithFiles.FileIds, draftResp.FileIds)
```

Comment:
Hmm.. I am a bit confused here, why does `draftResp.FileIds` has 2 file IDs, shouldn't we also checking if the file ID is *valid* during upsert?